dev-python/ipykernel/ipykernel-5.1.3.ebuild

```diff
@@ -0,0 +1,36 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
+PYTHON_REQ_USE="threads(+)"
+
+inherit distutils-r1
+
+DESCRIPTION="IPython Kernel for Jupyter"
+HOMEPAGE="https://github.com/ipython/ipykernel"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+distutils_enable_tests nose
```

Comment:
Great that you used the eclass to run tests, but there are missing test dependencies. First, nose_warnings_filters is missing and it looks like 5.1.3 adds additional tests from flaky and other packages. You can still use the distutils_enable_tests nose call and then define the others in BDEPEND:

e.g. BDEPEND="test? ( )"

Reply:
Fixed this in the new commit, I had to add IUSE="test" because the conditional test? is before the call to distutils_enable_tests, I could put it before the conditional, but then I'd have to put pytest back in DEPEND. Because distutils_enable_tests only appends to DEPEND if it is called after setting it in the ebuild.

Reply:
You only need to add the dependencies which distutils_enable_tests will not set. This is why distutils_enable_tests is called after DEPEND/RDEPEND/BDEPEND as it += on BDEPEND/RDEPEND/DEPEND etc.

Reply:
I think I did that correctly like this:

```
	test? (
		dev-python/nose[${PYTHON_USEDEP}]
		dev-python/nose_warnings_filters[${PYTHON_USEDEP}]
		dev-python/flaky[${PYTHON_USEDEP}] )"
```

I found this in the setup.py file on github:

```
    extras_require={
        'test': [
            'pytest',
            'pytest-cov',
            'flaky',
            'nose',  # nose because there are still a few nose.tools imports hanging around
        ],
```

The comment suggests that they have moved from using nose to pytest, therefore I have added `distutils_enable_tests pytest` to the ebuild instead of `distutils_enable_tests nose`. However, because nose is still listed in the setup.py file as a test dependency I kept it in DEPEND for now.